nextstrain/cli/command/build.py

```diff
@@ -306,10 +306,31 @@ def assert_overlay_volumes_support(opts):
     """
     overlay_volumes = opts.volumes
 
-    if overlay_volumes and opts.__runner__ not in {docker, singularity}:
+    if not overlay_volumes:
+        return
+
+    if opts.__runner__ not in {docker, singularity, aws_batch}:
         raise UserError(f"""
             The {runner_name(opts.__runner__)} runtime does not support overlays (e.g. of {overlay_volumes[0].name}).
-            Use the Docker or Singularity runtimes (via --docker or --singularity) if overlays are necessary.
+            Use the Docker, Singularity, or AWS Batch runtimes (via --docker,
+            --singularity, or --aws-batch) if overlays are necessary.
+            """)
+
+    if opts.__runner__ is aws_batch and not docker.image_supports(docker.IMAGE_FEATURE.aws_batch_overlays, opts.image):
+        raise UserError(f"""
+            The Nextstrain runtime image version in use
+
+                {opts.image}
+
+            is too old to support overlays (e.g. of {overlay_volumes[0].name}) with AWS Batch.
+
+            If overlays are necessary, please update the runtime image to at
+            least version
+
+                {split_image_name(opts.image)[0]}:{docker.IMAGE_FEATURE.aws_batch_overlays.value}
+
+            using `nextstrain update docker`.  Alternatively, use the Docker or
```

Comment:
Seeing `nextstrain update docker` for the `aws_batch` runner reminded me of https://github.com/nextstrain/cli/issues/328 😅 

Maybe include `--image <version>` in the error message as an alternative.

Reply:
Yeah, same… it'd be good to address that.

I intentionally left out `--image` to a) not further complicate the message and b) gently push people towards updating since periodically updating is more broadly preferable than slapping in `--image` at a fixed version and never ever updating it. I figure people already familiar with `--image` will realize they can use it too.

If you think otherwise, I'll add it in, though.

Reply:
(Probably should have written that in a comment in the code to start…)

Reply:
Yeah, it just feels a little weird to suggest a command that would not work for a subset of users that don't use the docker runtime locally. If the user does not use the docker runtime then I _think_ the only way they would have requested an outdated image is via `--image` or `NEXTSTRAIN_DOCKER_IMAGE`.

For not wanting to complicate the message, maybe just remove the "using `nextstrain update docker`." bit.

Reply:
Oh I see. I missed/forgot that it wasn't just weird but it'd actually just not work. Right.

Reply:
I want to either fully separate or fully unify the image versions used between runtimes (i.e. make AWS Batch uncoupled from Docker, or more tightly couple versions used by Docker/Singularity/AWS Batch), but since that's not happening immediately, I decided to [expand](https://github.com/nextstrain/cli/compare/86fb15446ad34d36e8655383ce926f46b5e41946..2837318f7ac44c1ad8fd65325076acef9c3eb11a) the error message a bit.

Thanks for helping me see the issue here!